agoda.py

```diff
@@ -0,0 +1,55 @@
+# import requests
+# import mysql.connector
+# import pandas as pd
+
+# Given a fixed-length integer array arr, duplicate each occurrence of zero, shifting the remaining elements to the right.
+
+# Note that elements beyond the length of the original array are not written. Do the above modifications to the input array in place
+
+# Input: arr = [1,0,2,3,0,4,5,0]
+# Output: [1,0,0,2,3,0,0,4]
+
+def sol(arr):
+    original_len = len(arr)
+    index = 0
+    while index < len(arr):
+        if arr[index] == 0:
+            # shift to right
+            arr[-1] = 0
+            end_ptr = original_len - 1
+            while end_ptr > index:
+                arr[end_ptr], arr[end_ptr - 1] = arr[end_ptr - 1], arr[end_ptr]
```

Comment:
```suggestion
                arr[end_ptr] = arr[end_ptr - 1]
```

What's the second assignment trying to achieve where when that value gets replaced in the next iteration anyways